zerver/lib/actions.py

```diff
@@ -3667,16 +3668,76 @@ def do_invite_users(user_profile, invitee_emails, streams, body=None):
         stream_ids = [stream.id for stream in streams]
         prereg_user.streams.set(stream_ids)
 
-        event = {"email": prereg_user.email, "referrer_id": user_profile.id, "email_body": body}
-        queue_json_publish("invites", event,
-                           lambda event: do_send_confirmation_email(prereg_user, user_profile, body))
+        queue_send_invitation_email(prereg_user, user_profile, body)
 
     if skipped:
         raise InvitationError(_("Some of those addresses are already using Zulip, "
                                 "so we didn't send them an invitation. We did send "
                                 "invitations to everyone else!"),
                               skipped, sent_invitations=True)
 
+def queue_send_invitation_email(prereg_user, user_profile, body):
+    event = {"email": prereg_user.email, "referrer_id": user_profile.id, "email_body": body}
+    queue_json_publish("invites", event,
+                           lambda event: do_send_confirmation_email(prereg_user, user_profile, body))
```

Comment:
Can you move this function extraction to be its own commit at the start of the branch?  We can then merge that now while we iterate with you on the rest of the PR, to help shrink the diff and avoid merge conflicts.